include/dlaf/util_matrix.h

```diff
@@ -441,6 +439,16 @@ void set_random_hermitian_with_offset(Matrix<T, Device::CPU>& matrix, const Size
   }
 }
 
+template <class T, Device D>
+dlaf::matrix::internal::SubMatrixSpec sub_matrix_spec_slice_cols(const Matrix<T, D>& matrix,
+                                                                 SizeType first_col_index,
+                                                                 SizeType last_col_index) {
+  DLAF_ASSERT(first_col_index >= 0, first_col_index);
+  DLAF_ASSERT(last_col_index < matrix.size().cols(), last_col_index, matrix.size().cols());
```

Comment:
This condition does not currently hold for zero-sized matrices (that we have in the tests), since `last_col_index` is inclusive and therefore the sub-matrix dimension is computed as `last_col_index - first_col_index + 1`. This is a good argument to make `last_col_index` exclusive as @rasolca suggested.

Reply:
Should be fixed by the new conditions, with `[begin, end)`.